src/ensembl/src/shared/state/ens-object/ensObjectActions.ts

```diff
@@ -17,107 +17,152 @@
 import { createAsyncAction } from 'typesafe-actions';
 import { Action } from 'redux';
 import { ThunkAction } from 'redux-thunk';
-import apiService from 'src/services/api-service';
+import { gql } from '@apollo/client';
+
+import { client } from 'src/gql-client';
 
 import { shouldFetch } from 'src/shared/helpers/fetchHelper';
 import {
   parseEnsObjectId,
   buildEnsObjectId,
   buildRegionObject,
-  EnsObjectIdConstituents
+  EnsObjectIdConstituents,
+  buildEnsObject
 } from './ensObjectHelpers';
 
 import { getGenomeExampleFocusObjects } from 'src/shared/state/genome/genomeSelectors';
 import { getEnsObjectLoadingStatus } from 'src/shared/state/ens-object/ensObjectSelectors';
 
-import { TrackId } from 'src/content/app/browser/track-panel/trackPanelConfig';
-import { EnsObject, EnsObjectResponse } from './ensObjectTypes';
+import { EnsObject } from './ensObjectTypes';
 import { RootState } from 'src/store';
+import { FullGene } from 'src/shared/types/thoas/gene';
 
 export const fetchEnsObjectAsyncActions = createAsyncAction(
   'ens-object/fetch_ens_object_request',
   'ens-object/fetch_ens_object_success',
   'ens-object/fetch_ens_object_failure'
 )<string, { id: string; data: EnsObject }, Error>();
 
-export const fetchEnsObject = (
-  payload: string | EnsObjectIdConstituents
-): ThunkAction<void, any, null, Action<string>> => async (
-  dispatch,
-  getState: () => RootState
-) => {
-  if (typeof payload === 'string') {
-    payload = parseEnsObjectId(payload);
-  }
-  const state = getState();
-  const ensObjectId = buildEnsObjectId(payload);
-  const ensObjectLoadingStatus = getEnsObjectLoadingStatus(state, ensObjectId);
-  if (!shouldFetch(ensObjectLoadingStatus)) {
-    return;
+const geneQuery = gql`
+  query Gene($genomeId: String!, $geneId: String!) {
+    gene(byId: { genome_id: $genomeId, stable_id: $geneId }) {
+      name
+      stable_id
+      unversioned_stable_id
+      symbol
+      slice {
+        region {
+          name
+        }
+        location {
+          start
+          end
+        }
+        strand {
+          code
+        }
+      }
+      metadata {
+        biotype {
+          label
+        }
+      }
+      transcripts {
+        stable_id
+        slice {
+          location {
+            length
+          }
+        }
+        product_generating_contexts {
+          product_type
+        }
+        metadata {
+          biotype {
+            label
+          }
+          canonical {
+            value
+            label
+          }
+          mane {
+            value
+            label
+            ncbi_transcript {
+              id
+              url
+            }
+          }
+          gencode_basic {
```

Comment:
why?

Reply:
That got added by mistake. I'll remove it.

Reply:
✅